index.src.html

```diff
@@ -1888,49 +1929,77 @@
 </pre>
 
 ### Internal Slots ###{#audioframe-internal-slots}
-<dl>
-  <dt><dfn attribute for=AudioFrame>\[[detached]]</dfn></dt>
-  <dd>
-    Boolean indicating whether close() was invoked and underlying resources
-        have been released.
-  </dd>
-</dl>
+: <dfn attribute for=AudioFrame>\[[detached]]</dfn>
+:: Boolean indicating whether {{AudioFrame/close()}} was invoked.
 
+: <dfn attribute for=AudioFrame>[[resource reference]]</dfn>
+:: A reference to a [=frame resource=] that stores the audio sample data for
+    this frame.
 
 ### Constructors ###{#audioframe-constructors}
 <dfn constructor for=AudioFrame title="AudioFrame(init)">
   AudioFrame(init)
 </dfn>
-1. Let |frame| be a new {{AudioFrame}} object.
-2. Assign `init.timestamp` to `frame.timestamp`.
-3. Assign `init.buffer` to `frame.buffer`.
-4. Assign `false` to the {{AudioFrame/[[detached]]}} internal slot.
-5. Return |frame|.
-
+1. Let |frame| be a new {{AudioFrame}} object, initialized as follows:
+    1. Assign |init|.{{AudioFrameInit/timestamp}} to {{AudioFrame/timestamp}}.
+    2. Assign |init|.{{AudioFrameInit/buffer}} to {{AudioFrame/buffer}}.
+    3. Let |resource| be a new [=frame resource=].
+    4. [=Acquire the content=] [[WEBAUDIO]] of {{AudioFrame/buffer}} and
+        assign the acquired content to |resource|.
+
+        NOTE: The acquired content stored in |resource| is an immutable
+            snapshot {{AudioFrameInit/buffer}}'s content. Mutating
+            {{AudioFrame/buffer}} after construction (e.g. via
+            {{AudioBuffer/copyToChannel()}}) will have no effect on the
+            |resource|.
```

Comment:
I think you can more simply state that assigning `AudioFrameInit.buffer` to `AudioFrame.buffer` "acquires the content" of `buffer`, and that's it, not that you assign the "acquired content". But it might be more flexible to do this "acquire the content" operation only when calling `decode()` with a particular `AudioFrame`: only then it's actually sent to another thread and authors must be unable to tamper with it, etc. This would be consistent with what the Web Audio API does.

Reply:
I'm of two minds on this one...

Mind A:
On the call we discussed whether this might move to encode(). I mentioned that this idea works but it breaks symmetry with other IO types (VideoFrame and Encoded*Chunk) which are immutable. This PR is otherwise making the two interfaces even more symmetrical. I think symmetry makes the API more intuitive.

Mind B:
Having an "immutable" AudioFrame who's buffer appears to be mutable (e.g. copyFromChannel()) is not very intuitive. We could instead "acquire the content" at `encode()` and otherwise let folks mutate their AudioBuffer all they like. If we pursue this, I'll want to remove `AudioFrame.clone()` entirely. The AudioFrame never acquired the content, so there isn't a "frame resource" to refcount. The content is owned by AudioBuffer. If folks want to take additional JS references on AudioBuffer, that is easy enough. We could continue to define `AudioFrame.close()`, which just nulls out the buffer giving us a chance for immediate cleanup rather than waiting for GC. Even that seems unimportant.

Having written that, I guess I'm leaning toward option 'B' as being less surprising and more flexible. @hoch @rtoy @dalecurtis @sandersdan - any concerns? 

If we take this route, I would drop this PR and move the entire frame resource / refcounting concept into the VideoFrame PR, making it VideoFrame-specific.

Reply:
I'm wondering if we even want buffer as part of the AudioFrameInit structure. Since we allow VideoFrame(CanvasImageSource, VideoFrameInit) maybe it should be AudioFrame(buffer, optional AudioFrameInit).

If we want this to work similar to the VideoFrame mechanism than it is indeed taking a Snapshot in time of |buffer|, at the moment of creation buffer becomes immutable henceforth. I think the interface benefits from symmetry here.

Reply:
> I'm wondering if we even want buffer as part of the AudioFrameInit structure. Since we allow VideoFrame(CanvasImageSource, VideoFrameInit) maybe it should be AudioFrame(buffer, optional AudioFrameInit).

I think VideoFrame is the weird one actually. The constructor shape made more sense when we were using the same init dicts between the two constructors, but now that we've split those I lean toward putting even more things into the distinct VideoFrameInit and VideoFramePlanesInit dicts, making both constructors take a single argument. But this is out of scope for this CL IMO.

> If we want this to work similar to the VideoFrame mechanism than it is indeed taking a Snapshot in time of |buffer|, at the moment of creation buffer becomes immutable henceforth. I think the interface benefits from symmetry here.

Ah, I forgot we were taking snapshots there too. This puts me somewhat at ease. With that + your edit, I'm back to favoring the proposal as written.

Reply:
Discussed on editors call. Resolved to go ahead w/ plans to break from AudioBuffer as described in https://github.com/w3c/webcodecs/issues/179